src/index.ts

```diff
@@ -25,15 +25,36 @@ type Preferences = {
   squashMergeAllowed?: boolean,
 }
 
-const getToken = async () => {
+// Prompt the user for a token and return it as a string
+const tokenPrompt = async () => {
   console.log('Create a token at https://github.com/settings/tokens/new?scopes=repo')
-  return (await prompts({
+  const response = await prompts({
     type: 'password',
     name: 'token',
     message: 'GitHub access token:',
-    validate: (value: string) => value.startsWith('ghp_') ? true : 'Your token should start with ghp_',
-  })).token;
-}
+    validate: (value: String) =>
+      value.startsWith('ghp_') ? true : 'Your token should start with ghp_',
+  });
+  return response.token;
+};
+
+// Prompt the user with a choice to use the GITHUB_TOKEN from their environment, or to enter a token
+const getToken = async () => {
+  const { token } = await prompts({
+    type: 'select',
+    name: 'token',
+    message: 'How would you like to authenticate?',
+    choices: [
+      { title: 'Use GITHUB_TOKEN from environment', value: process.env.GITHUB_TOKEN },
```

Comment:
Maybe we should show this option (and entire question?) if there is a string beginning `ghp_` in `process.env.GITHUB_TOKEN`?

Cost of this as a proposed approach is that it reduces the discoverability of this as an option, but I think that's an okay trade-off. We could always add a log line somewhere else along the lines of `Tip: Want to avoid entering your token next time? You can set it in the environment variable GITHUB_TOKEN.`

Reply:
Good advice, thank you - implemented this one.